# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/ipc/NeckoCommon.h
# Commit: aea7ca0bbd17
# Full Hash: aea7ca0bbd17e31625ac66bbb1815858ff135c35
# Author: Manuel Bucher <manuel@mozilla.com>
# Date: 2022-12-02 21:26:33
# Regressor Bug: 1771867
# File Overlap Count: 1
# Description:
#   Bug 1771867 - Early Hints Phase 2 - Part 12: Redirect EarlyHint preload channel to content process r=necko-reviewers,kershaw,valentin
#   
#   Similar to how DocumentLoadListener redirects the document to the content process.
#   
#   This fixes the race conditions in early hint architecture fixing intermittent
# ==============================================================================

diff -r 2fe2afdee09d -r aea7ca0bbd17 netwerk/ipc/NeckoCommon.h
--- a/netwerk/ipc/NeckoCommon.h	Fri Dec 02 09:45:28 2022 +0000
+++ b/netwerk/ipc/NeckoCommon.h	Fri Dec 02 09:45:28 2022 +0000
@@ -15,6 +15,8 @@
 #include "nsXULAppAPI.h"
 #include "prenv.h"
 
+class nsIStreamListener;
+
 namespace mozilla {
 namespace dom {
 class BrowserChild;
@@ -132,6 +134,9 @@
     mozilla::Variant<OnStartRequestParams, OnDataAvailableParams,
                      OnStopRequestParams, OnAfterLastPartParams>;
 
+nsresult ForwardStreamListenerFunctions(
+    nsTArray<StreamListenerFunction>& aCalls, nsIStreamListener* aParent);
+
 }  // namespace net
 }  // namespace mozilla
 